libraries/protocol/smt_operations.cpp

```diff
@@ -192,7 +192,42 @@ void smt_cap_reveal_operation::validate()const {}
 void smt_refund_operation::validate()const {}
 void smt_setup_inflation_operation::validate()const {}
 void smt_set_setup_parameters_operation::validate()const {}
-void smt_set_runtime_parameters_operation::validate()const {}
+
+struct smt_set_runtime_parameters_operation_visitor
+{
+   smt_set_runtime_parameters_operation_visitor(){}
+
+   typedef void result_type;
+
+   void operator()( const smt_param_windows_v1& param_windows ) const
+   {
+      FC_ASSERT(  ( param_windows.cashout_window_seconds > ( param_windows.reverse_auction_window_seconds + SMT_UPVOTE_LOCKOUT ) ) &&
+                  ( param_windows.cashout_window_seconds < SMT_VESTING_WITHDRAW_INTERVAL_SECONDS ), "Cashout window second must be valid" );
+
+      FC_ASSERT( ( param_windows.reverse_auction_window_seconds + SMT_UPVOTE_LOCKOUT ) >= 0, "Reverse auction window seconds must be valid" );
```

Comment:
Again, I feel like error messages should be descriptive. this assertion's text is a repetition of the assertion itself.